modules/tinymce/src/plugins/fullscreen/test/ts/browser/FullScreenPluginTest.ts

```diff
@@ -1,17 +1,31 @@
 import { Assertions, Chain, Guard, Log, Pipeline, UiFinder } from '@ephox/agar';
 import { UnitTest } from '@ephox/bedrock-client';
 import { Cell } from '@ephox/katamari';
-import { Editor as McEditor } from '@ephox/mcagar';
-import { Attribute, Html, SelectorFind, SugarBody, SugarElement } from '@ephox/sugar';
+import { Editor as McEditor, TinyLoader } from '@ephox/mcagar';
+import { Attribute, Classes, Css, Html, SelectorFind, SugarBody, SugarElement, SugarShadowDom } from '@ephox/sugar';
 import Editor from 'tinymce/core/api/Editor';
 
 import FullscreenPlugin from 'tinymce/plugins/fullscreen/Plugin';
 import LinkPlugin from 'tinymce/plugins/link/Plugin';
 import SilverTheme from 'tinymce/themes/silver/Theme';
 
-export const cWaitForDialog = (ariaLabel) =>
+const getContentContainer = (editor: Editor) =>
+  SugarShadowDom.getContentContainer(SugarShadowDom.getRootNode(SugarElement.fromDom(editor.getElement())));
+
+const cCloseOnlyWindow = Chain.control(
+  Chain.op((editor: Editor) => {
+    const dialogs = () => UiFinder.findAllIn(getContentContainer(editor), '[role="dialog"]');
+    Assertions.assertEq('One window exists', 1, dialogs().length);
+    editor.windowManager.close();
+    Assertions.assertEq('No windows exist', 0, dialogs().length);
+  }),
+  Guard.addLogging('Close window')
+);
+
+const cWaitForDialog = (ariaLabel: string) =>
   Chain.control(
-    Chain.fromChainsWith(SugarBody.body(), [
+    Chain.fromChains([
+      Chain.mapper(getContentContainer),
```

Comment:
If I'm reading this right, this is dependent on running after a Chain that happens to pass along `editor`? I'm wondering if it would be safer/easier to read if we inject `editor` in case... Not sure though.